src/components/ObsDetails/FlagItemModal.js

```diff
@@ -0,0 +1,178 @@
+      if ( value === "other" ) {
+        params = { ...params, flag_explanation: explanation };
+      }
+      createFlagMutation.mutate( params );
+    }
+  };
+
+  return (
+    <Modal
+      visible={showFlagItemModal}
+      animationType="slide"
+      className="flex-1"
+    >
+      <SafeAreaView className="flex-1">
+        <View className="flex-row-reverse justify-between p-6 border-b">
+          <IconMaterial name="close" onPress={closeFlagItemModal} size={30} />
+          <Text className="text-xl">
+            {t( "Flag-An-Item" )}
+          </Text>
+        </View>
+        <KeyboardAwareScrollView
+          ref={keyboardScrollRef}
+          enableOnAndroid
+          enableAutomaticScroll
+          extraHeight={200}
+          className="p-6"
+        >
+          <Text className="text-base">
+            {t( "Flag-Item-Description" )}
+          </Text>
+          <View className="flex-row my-2">
+            <CheckBox
+              disabled={false}
+              value={value === "spam"}
+              onValueChange={() => toggleValue( "spam" )}
+            />
+            <Text
+              className="font-bold text-lg ml-5"
+              onPress={() => toggleValue( "spam" )}
+            >
+              {t( "Spam" )}
+
+            </Text>
+          </View>
+          <Text className="mb-2 text-base" style>{t( "Spam-Examples" )}</Text>
+
+          <View className="flex-row my-2">
+            <CheckBox
+              disabled={false}
+              value={value === "inappropriate"}
+              onValueChange={() => toggleValue( "inappropriate" )}
+            />
+            <Text
+              className="font-bold text-lg ml-5"
+              onPress={() => toggleValue( "inappropriate" )}
+            >
+              {t( "Offensive-Inappropriate" )}
+
+            </Text>
+          </View>
+          <Text className="mb-2 text-base">{t( "Offensive-Inappropriate-Examples" )}</Text>
+
+          <View className="flex-row my-2">
+            <CheckBox
+              disabled={false}
+              value={value === "other"}
+              onValueChange={() => toggleValue( "other" )}
+            />
+            <Text
+              className="font-bold text-lg ml-5"
+              onPress={() => toggleValue( "other" )}
+            >
+              {t( "Other" )}
+
+            </Text>
+          </View>
+          <Text className="mb-2 text-base">{t( "Flag-Item-Other-Description" )}</Text>
+          {( value === "other" )
+            ? (
+              <>
+                <TextInput
+                  className="text-sm"
+                  placeholder={t( "Flag-Item-Other-Input-Hint" )}
+                  value={explanation}
+                  onChangeText={text => setExplanation( text )}
+                  onFocus={e => scrollToInput( findNodeHandle( e.target ) )}
+                />
+                <Text>{`${explanation.length}/255`}</Text>
+              </>
+            )
+            : undefined}
```

Comment:
Instead of using a tertiary statement and `undefined` here, you can use && to only show the component if the value equals "other".